aiounifi/models/api.py

```diff
@@ -27,6 +29,22 @@ def full_path(self, site: str, is_unifi_os: bool) -> str:
             return f"/proxy/network/api/s/{site}{self.path}"
         return f"/api/s/{site}{self.path}"
 
+    def prepare_data(self, raw: bytes) -> TypedApiResponse:
+        """Put data, received from the unifi controller, into a TypedApiResponse."""
+        json_data = orjson.loads(raw)
+        return_data: TypedApiResponse = {}
+
+        if isinstance(json_data, dict):
+            if "meta" in json_data:
+                return_data["meta"] = json_data.get("meta", {})
+                return_data["data"] = json_data.get("data", [])
+            else:
+                return_data["data"] = [json_data]
+        if isinstance(json_data, list):
+            return_data["data"] = json_data
+
+        return return_data
```

Comment:
I think the default behaviour should be the old one. Then you should override this method within `class TrafficRuleRequest(ApiRequest):` When we get more v2 apis that follow the same structure then a ApiRequestV2 could be provided as a base class.

Reply:
Ok, I have moved everything that is special for the v2 api down to the TrafficRuleRequest(ApiRequest). Some little changes were still needed in other parts for working with a TypedApiResponse. I'm still finetuning some things before committing.

Reply:
Some extra conditions seemed to be required (the tests pointed me in that direction). They are added in the code.